src/Core/Framework/Plugin/PluginLifecycleService.php

```diff
@@ -508,4 +528,25 @@ private function getPluginInstance(string $pluginBaseClassString): Plugin
 
         return $this->getPluginBaseClass($pluginBaseClassString);
     }
+
+    private function savePluginConfiguration(Plugin $plugin, InstallContext $context): void
+    {
+        $config     = $this->configReader->getConfigFromBundle($plugin);
+        $prefix     = $plugin->getName() . '.config.';
+        $noOverride = $context instanceof UpdateContext;
+
+        foreach ($config as $card) {
+            foreach ($card['elements'] as $element) {
+                $key   = $prefix . $element['name'];
+                $value = $element['defaultValue'];
+
+                if (
```

Comment:
please add some tests, which checks the different cases with different options

Reply:
Do you mean asserting the values which results by the type-cast (see conversation above)?

Reply:
a test with a plugin with config with default values, which gets installed and after that you try to read the value with the ConfigService and it has to be a certain value

Reply:
I have added tests.